src/VideoComponents/MiniMapWidget.jsx

```diff
@@ -1,147 +1,82 @@
-    ctx.strokeStyle = '#FFFFFF';
-    ctx.stroke();
-
-    //draw boat
-    ctx.save(); //save context state so we can draw boat and ROV from different origins and rotate independently
-    ctx.translate(mapWidth / 2, mapWidth / 2); //draw boat from the middle of the circle
-    //ctx.rotate(boatAngle[0]); //rotates the boat drawing about the middle of the circle
-    drawBoat(ctx, boatWidth, boatLength, boatHeading);
-    ctx.restore(); //restore context state we saved earlier
-
-    const inBoundsEast =
-      east <= maxDist * rightFactor && east >= -maxDist * leftFactor; //multiply by left/rightFactor so the ROV square won't be replaced by arrow too early or late
-    const inBoundsNorth = Math.abs(north) <= maxDist;
-    //draw ROV
-    ctx.save();
-    ctx.translate(mapWidth / 2, mapWidth / 2);
-    ctx.rotate(-boatRotation); //rotates ROV around boat when boat rotates
+    // This is necessary to make 'this' work inside the zoom functions
+    this.zoomOut = this.zoomOut.bind(this);
+    this.zoomIn = this.zoomIn.bind(this);
+  }
 
-    if (inBoundsEast && inBoundsNorth) {
-      //draw ROV as a square within the map
-      const mapNorth = mapRange(
-        north,
-        -maxDist,
-        maxDist,
-        -mapWidth / 2,
-        mapWidth / 2,
-      ); //map north prop to the pixel range we're working with
-      const mapEast = mapRange(
-        east,
-        -maxDist,
-        maxDist,
-        -mapWidth / 2,
-        mapWidth / 2,
-      ); //map east prop to the pixel range we're working with
-      ctx.save();
-      ctx.translate(boatWidth + mapEast + rovSize / 2, -mapNorth); //draw square at correct point in the map
-      ctx.rotate(yaw);
-      drawROV(ctx, rovSize);
-      ctx.restore();
-    } else {
-      //draw arrow pointing to where the ROV is outside the map
-      ctx.save();
-      ctx.translate(0, 0); //origin in middle of map so we can rotate arrow about this point
-      ctx.rotate(-ROVangleInNED); //point/rotate arrow in direction of ROV
-      drawArrow(
-        ctx,
-        rovSize *
-          (maxDist /
-            Math.log(Math.exp(maxDist) + Math.pow(Math.max(east, north), 3))),
-        mapWidth,
-      ); //make size of arrow based on how far out of bounds the ROV is
-      ctx.restore();
-    }
-    ctx.fillStyle = '#00FF00';
-    ctx.fill();
+  componentDidMount() {
+    initMinimap(0);
+    super.componentDidMount();
+  }
 
-    ctx.translate(boatWidth + rovSize / 2, 0);
-    drawNEDframe(ctx, mapWidth, boatWidth, rovSize);
-    ctx.restore();
-  }, [boatHeading, east, north, yaw, maxDist, boatRotation]);
+  componentDidUpdate() {
+    drawMinimap(
+      this.ctx,
+      this.props.north,
+      this.props.east,
+      this.props.yaw,
+      this.props.boatHeading,
+      this.maxDistance,
+      initialWidth,
+      initialHeight,
+    );
+  }
 
-  function zoomOut() {
-    if (maxDist < 1000) {
-      //maximum 1000 metres
-      setMaxDist(maxDist + 1);
+  zoomOut() {
```

Comment:
I think if you write const zoomOut = () => {....
then you won't need to bind like you have done above. I might be wrong, and it is not important